subprojects/logging/src/main/java/org/gradle/internal/featurelifecycle/FeatureUsage.java

```diff
@@ -27,43 +27,63 @@
  * An immutable description of the usage of a deprecated feature.
  */
 public class FeatureUsage {
-    private final String message;
-    private final String details;
+    private final String summary;
+    private final String removalDetails;
     private final String advice;
+    private final FeatureUsageType type;
     private final Class<?> calledFrom;
     private final Exception traceException;
+    private final String contextualAdvice;
 
     private List<StackTraceElement> stack;
 
-    public FeatureUsage(String message, String details, String advice, Class<?> calledFrom) {
-        this.message = message;
-        this.details = details;
+    public enum FeatureUsageType {
+        USER_CODE_DIRECT,
+        USER_CODE_INDIRECT,
+        INVOCATION
+    }
```

Comment:
What the meaning of these types? Seems like they're not used in Gradle codebase, are they used by build scan somewhere?

Reply:
fixed